plugin/trino-hive/src/test/java/io/trino/plugin/hive/parquet/TestParquetDecimalScaling.java

```diff
@@ -463,6 +508,13 @@ private static String generateTableName(String testCase, int precision, int scal
         return format("%s_%d_%d_%d", testCase, precision, scale, ThreadLocalRandom.current().nextInt(1, MAX_VALUE));
     }
 
+    private static Object[][] withWriterVersion(Object[][] args)
+    {
+        return Arrays.stream(cartesianProduct(args, WriterVersion.values()))
+                .map(TestNGUtils::combineAndFlatten)
```

Comment:
The above doesn't work because the the object array of the `args` without writer version needs to be flattened. I've now removed `combineAndFlatten` and added a utility function in this class for that.

Reply:
Right, because the signature of `cartesianProduct` should be `public static Object[][] cartesianProduct(Object[][]... args)`, since it is supposed to work with data providers.

Reply:
I've updated cartesianProduct to accept `Object[][]... args` now